src/InteractiveWindow/Editor/InteractiveWindow.UIThreadOnly.cs

```diff
@@ -1378,6 +1403,11 @@ private void RemoveLastInputPrompt()
             private void AddLanguageBuffer()
             {
                 ITextBuffer buffer = _factory.CreateAndActivateBuffer(_window);
+                if (CurrentLanguageBuffer != null)
+                {
+                    _textBufferUndoManagerProvider.RemoveTextBufferUndoManager(CurrentLanguageBuffer);
+                }
+                UndoManager = _textBufferUndoManagerProvider.GetTextBufferUndoManager(buffer);
```

Comment:
Traditionally buffer managers map to the surface buffer and not subject buffers. I could understand that this scenario might be different. You've carefully thought about this?